rubix-common/src/main/java/com/qubole/rubix/common/utils/ClusterUtil.java

```diff
@@ -50,4 +56,33 @@ public static String getMasterHostname(Configuration conf)
     log.debug("No hostname found in etc/*-site.xml, returning localhost");
     return "localhost";
   }
+
+  public static synchronized Configuration applyRubixSiteConfig(Configuration conf)
+  {
+    if (CacheConfig.isRubixSiteConfigApplied(conf)) {
+      return conf;
+    }
+
+    if (rubixSiteExists.get() == null) {
+      synchronized (rubixSiteExists) {
```

Comment:
the method is already synchronized, why do we need to synchronize here?

Reply:
This was my bad, just copied the signature as is in my previous comment. We should synchronize the whole method

Reply:
fixed